src/main/java/org/testng/internal/MethodRunner.java

```diff
@@ -0,0 +1,134 @@
+package org.testng.internal;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.testng.ITestContext;
+import org.testng.ITestResult;
+import org.testng.collections.CollectionUtils;
+import org.testng.collections.Lists;
+import org.testng.internal.ITestInvoker.FailureContext;
+import org.testng.internal.TestMethodArguments.Builder;
+import org.testng.xml.XmlSuite;
+
+class MethodRunner implements IMethodRunner {
+
+  @Override
+  public List<ITestResult> runInSequence(TestMethodArguments arguments,
+      ITestInvoker testInvoker,
+      ITestContext context,
+      AtomicInteger invocationCount,
+      FailureContext failure,
+      Iterator<Object[]> allParamValues,
+      boolean skipFailedInvocationCounts) {
+    List<ITestResult> result = Lists.newArrayList();
+    int parametersIndex = 0;
+    Iterable<Object[]> allParameterValues = CollectionUtils.asIterable(allParamValues);
```

Comment:
In fact, I supposed that it was possible to change parameter type but I like your idea too :)